onnxruntime/core/providers/cpu/reduction/reduction_ops.cc

```diff
@@ -199,18 +200,37 @@ bool PrepareForReduce(OpKernelContext* ctx,
   int64_t first_dim = 1;
   std::vector<int64_t> reduced_dims;
```

Comment:
slightly outside the scope of this PR. But I guess we could reserve memory before the `push_back`s below

Reply:
Added.

Reply:
might be worth just caching in_dims.size() and re-using in the next line, but maybe in another PR...